### ISSUE TITLE
aiohttp failing exchange api call when setting proxy

### ISSUE THREAD

**Issue by FreekyDev** [ccxt/ccxt#18349]
### Operating System

debian (Docker python3.11-slim-bullseye)

### Programming Languages

python

### CCXT Version

3.1.52

### Description

>File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/base/exchange.py", line 251, in load_markets
    raise e
  File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/base/exchange.py", line 247, in load_markets
    result = await self.markets_loading
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/base/exchange.py", line 237, in load_markets_helper
    markets = await self.fetch_markets(params)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/binance.py", line 1928, in fetch_markets
    promises = await asyncio.gather(*promisesRaw)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/binance.py", line 7451, in request
    response = await self.fetch2(path, api, method, params, headers, body, config)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/base/exchange.py", line 1823, in fetch2
    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ccxt/async_support/base/exchange.py", line 176, in fetch
    async with session_method(yarl.URL(url, encoded=True),
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
                 ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 899, in _create_connection
    _, proto = await self._create_proxy_connection(req, traces, timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1325, in _create_proxy_connection
    return await self._start_tls_connection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 1124, in _start_tls_connection
    tls_proto.connection_made(
  File "/usr/local/lib/python3.11/site-packages/aiohttp/base_protocol.py", line 62, in connection_made
    tcp_nodelay(tr, True)
  File "/usr/local/lib/python3.11/site-packages/aiohttp/tcp_helpers.py", line 25, in tcp_nodelay
    sock = transport.get_extra_info("socket")
           ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get_extra_info'

I tested the same code in a test file (Windows 10 - Python 3.11) and it works. However in docker (debian) it gives me this error.

### Code

```
import asyncio
from ccxt.pro import binance as ccxt_binance

async def main():
    binance = ccxt_binance()
    binance.newUpdates = True
    binance.enableRateLimit = False
    binance.apiKey = 'XX'
    binance.secret = 'XX'
    binance.aiohttp_proxy = 'XX'
    await binance.load_markets()

asyncio.run(main())
```

**Comment by carlosmiei** [comment]
Hello @FreekyDev, 
Thanks for reporting it. Looks like a known issue: https://github.com/aio-libs/aiohttp/issues/6239 but we will try to find a workaround.

cc/ @ttodua

**Comment by FreekyDev** [comment]
> Hello @FreekyDev, Thanks for reporting it. Looks like a known issue: [aio-libs/aiohttp#6239](https://github.com/aio-libs/aiohttp/issues/6239) but we will try to find a workaround.
> 
> cc/ @ttodua

Yes, I've seen it is a known issue, however I tested it on Windows 10 and works. Just wanted to point it out since i haven't seen any reports about it working on Windows and it might be helpful to know. Hope you guys will find a solution soon, thank you!

**Comment by ttodua** [comment]
@FreekyDev in the meantime, is it possible you switched using to unified proxies (instead of aiohttp proxy?) so, you could use like shown here: https://github.com/ccxt/ccxt/blob/master/examples/py/proxy-usage.py (example 2)

**Comment by FreekyDev** [comment]
> @FreekyDev in the meantime, is it possible you switched using to unified proxies (instead of aiohttp proxy?) so, you could use like shown here: https://github.com/ccxt/ccxt/blob/master/examples/py/proxy-usage.py (example 2)

I tried
`binance.https_proxy = 'XX'`
Outputs same error as above

**Comment by FreekyDev** [comment]
> @FreekyDev what is your version of aiohttp ? Do you have this issue with the most recent versions of CCXT and aiohttp?

aiohttp: 3.8.4
ccxt: 3.1.54

**Comment by FreekyDev** [comment]
```
import asyncio
import aiohttp

async def main():
    async with aiohttp.ClientSession() as session:
        async with session.get('http://python.org',
                               proxy='http://xmcllhbz-rotate:sxb3s141bggm@p.webshare.io:80') as response:
            res = await response.text()
            print(res)

asyncio.run(main())
```
I have tested a simple code fragment with aiohttp==3.8.4 and it fails. The same test is successful with aiohttp==3.7.
Is it possible for you guys to temporarily set aiohttp version to >=3.7 in [https://github.com/ccxt/ccxt/blob/master/python/setup.py](url) until it gets fixed in 3.8?

**Comment by ttodua** [comment]
@carlosmiei about the above comment, wdyt? I don't have much context about the past of aiohttp related things in ccxt, do you see it feasible to do the above change?

**Comment by carlosmiei** [comment]
@FreekyDev I managed to reproduce it now by using an authenticated proxy :/

**Comment by FreekyDev** [comment]
@carlosmiei Recently aiohttp 3.8.5 came out, claiming it solves this problem. Can you give it a try?

**Comment by carlosmiei** [comment]
@FreekyDev Thanks for letting me know, will try that out

**Comment by carlosmiei** [comment]
@FreekyDev Interestingly 3.7.4 worked for me whereas 3.8.5 did not 🤔

**Comment by FreekyDev** [comment]
> Python 3.11
> 
> ```
> 
> import asyncio
> 
> import aiohttp
> 
> 
> 
> async def main():
> 
>     async with aiohttp.ClientSession() as session:
> 
>         async with session.get('http://python.org',
> 
>                                proxy='XX') as response:
> 
>             res = await response.text()
> 
>             print(res)
> 
> 
> 
> asyncio.run(main())
> 
> ```
> 
> I have tested a simple code fragment with aiohttp==3.8.4 and it fails. The same test is successful with aiohttp==3.7.
> 
> Is it possible for you guys to temporarily set aiohttp version to >=3.7 in [https://github.com/ccxt/ccxt/blob/master/python/setup.py](url) until it gets fixed in 3.8?

@carlosmiei Unfortunate it doesn't work on 3.8.5

**Comment by ttodua** [comment]
Hi all,
you can update ccxt to latest version, where proxies functionality should be fixed and also added Websockets support. 
For more info, see: https://docs.ccxt.com/#/?id=supported-proxy-types
